mne/stats/tests/test_cluster_level.py

```diff
@@ -911,121 +911,77 @@ def test_new_cluster_api(Inst):
+                _cond.data += rng.normal(scale=0.1, size=_cond.data.shape)
+                insts.append(_cond)
+        conds = np.repeat(["a", "b"], n_epo).tolist()
+    else:
+        # For Epochs(TFR)Array, each epoch is an observation and they're already
+        # noisy/non-identical, so no duplication / noise-addition necessary.
+        insts = [cond_a, cond_b]
         conds = ["a", "b"]
 
-    df = pd.DataFrame(dict(data=data, condition=conds))
-
+    # run new clustering API
+    df = pd.DataFrame(dict(data=insts, condition=conds))
     kwargs = dict(
         n_permutations=100, seed=42, tail=1, buffer_size=None, out_type="mask"
     )
-
     result_new_api = cluster_test(df, "data~condition", **kwargs)
 
     # make sure channels are last dimension for old API
-    if Inst == EpochsArray:
-        inst1 = inst1.get_data().transpose(0, 2, 1)
-        inst2 = inst2.get_data().transpose(0, 2, 1)
-    elif Inst == EpochsTFRArray:
-        inst1 = inst1.data.transpose(0, 3, 2, 1)
-        inst2 = inst2.data.transpose(0, 3, 2, 1)
-    elif Inst == AverageTFRArray:
-        inst1 = inst1.data.transpose(2, 1, 0)
-        inst2 = inst2.data.transpose(2, 1, 0)
-        inst1_n = inst1_n.data.transpose(2, 1, 0)
-        inst2_n = inst2_n.data.transpose(2, 1, 0)
-        # combine the data of the two subjects
-        inst1 = np.concatenate([inst1[np.newaxis, :], inst1_n[np.newaxis, :]], axis=0)
-        inst2 = np.concatenate([inst2[np.newaxis, :], inst2_n[np.newaxis, :]], axis=0)
+    if is_epo:
+        axes = (0, 3, 2, 1) if is_tfr else (0, 2, 1)
+        X = [cond_a.get_data().transpose(*axes), cond_b.get_data().transpose(*axes)]
     else:
-        inst1 = inst1.data.transpose(1, 0)
-        inst2 = inst2.data.transpose(1, 0)
-        inst1_n = inst1_n.data.transpose(1, 0)
-        inst2_n = inst2_n.data.transpose(1, 0)
-        # combine the data of the two subjects
-        inst1 = np.concatenate([inst1[np.newaxis, :], inst1_n[np.newaxis, :]], axis=0)
-        inst2 = np.concatenate([inst2[np.newaxis, :], inst2_n[np.newaxis, :]], axis=0)
-
-    F_obs, clusters, cluster_pvals, H0 = permutation_cluster_test(
-        [inst1, inst2], **kwargs
-    )
+        axes = (2, 1, 0) if is_tfr else (1, 0)
+        Xa = list()
+        Xb = list()
+        for inst, cond in zip(insts, conds):
+            container = Xa if cond == "a" else Xb
+            container.append(inst.get_data().transpose(*axes))
+        X = [np.stack(Xa), np.stack(Xb)]
+
+    F_obs, clusters, cluster_pvals, H0 = permutation_cluster_test(X, **kwargs)
     assert_array_almost_equal(result_new_api.H0, H0)
     assert_array_almost_equal(result_new_api.stat_obs, F_obs)
     assert_array_almost_equal(result_new_api.cluster_p_values, cluster_pvals)
-    assert result_new_api.clusters == clusters
+    assert len(result_new_api.clusters) == len(clusters)
+    for clu1, clu2 in zip(result_new_api.clusters, clusters):
+        assert_array_equal(clu1, clu2)
```

Comment:
here we can't use `==` because "the truth value of an array is ambiguous", so we check the lengths of the outer lists are the same, then assert that each array within the list matches.